### PR TITLE
Add commandbar browser destination

### PR DESCRIPTION
**Pull request by tkainrad**
## CommandBar Browser Destination Overview

We are thrilled to submit our new device mode browser destination. [CommandBar](https://www.commandbar.com/) provides a set of widgets to eliminate UX friction.

With this destination, we want to enable users to 

- initialize CommandBar
- add user `traits` to CommandBar on `identify` -> `identifyUser` action
     - this will allow users to control CommandBar behaviour based on these tratis 
- pass `track` calls CommandBar -> `trackEvent` action
    - this will let users trigger specific CommandBar behaviour like showing nudges

### Settings

```
{
    "orgId": "84f6747a",
    "deploy": true,
    "subscriptions": [{
            "name": "Identify User",
            "subscribe": "type = \"identify\"",
            "partnerAction": "identifyUser",
            "enabled": true,
            "mapping": {
                "userId": {
                    "@path": "$.userId"
                },
                "traits": {
                    "@path": "$.traits"
                },
                "hmac": {
                    "@path": "$.context.CommandBar.hmac"
                },
                "formFactor": {
                    "@path": "$.context.CommandBar.formFactor"
                }
            }
        },
        {
            "partnerAction": "trackEvent",
            "name": "Track Event",
            "enabled": true,
            "subscribe": "type = \"track\"",
            "mapping": {
                "event_name": {
                    "@path": "$.event"
                },
                "event_metadata": {
                    "@path": "$.properties"
                }
            }
        }
    ]
}
```

### Events

**Identify**

```json
{
    "userId": "test-base-user",
    "traits": {
        "foo": "bar"
    },
    "options": {
        "CommandBar": {
            "hmac": "cc9185c5d19062c92798472edec639ee20826aa048b8aa77fe868f556c849d93"
        }
    }
}
```

**Track**

```json
{
    "name": "test-event",
    "properties": {
        "foo": "bar"
    }
}
```

image

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

---

Let me know if you have any questions.

Thanks for creating this framework with its powerful testing environment; it has been great to work with!  
One minor feedback item: The local development server does not work with OpenSSL 3 which comes with the most recent Ubuntu versions.

### CODE REVIEW

**Review by lchoward — COMMENTED** [review]
@tkainrad this is awesome, played around with it using Segment's testing tools that you linked.

~Only major item I am wondering about: would it be possible/easy for us to add hmac in the scenario of `Deploy via Segment = false`? Could be a sweet addition out of the box.~
[EDIT: tested and works like a charm [https://share.commandbar.com/HmumYv](https://share.commandbar.com/HmumYv)]

A few other observations
- When testing via `./bin/run serve --directory ./packages/browser-destinations/src/destinations --browser`
  - ✅  Looks good on identifyUser (https://share.commandbar.com/5KbhvP)
  - ✅ Everything looks good for Segment's trackEvent (https://share.commandbar.com/m3D0fG)
- I couldn't figure out how to use the old testing method to load CB, might be worth doing if easy `./bin/run serve --directory ./packages/browser-destinations/src/destinations --browser`.
- Made a few minor copy nits

**Comment by tkainrad** [comment]
> @tkainrad this is awesome, played around with it using Segment's testing tools that you linked.
> 
> Only major item I am wondering about: would it be possible/easy for us to add hmac in the scenario of `Deploy via Segment = false`? Could be a sweet addition out of the box.
> 
> A few other observations
> 
> * When testing via `./bin/run serve --directory ./packages/browser-destinations/src/destinations --browser`
>   
>   * white_check_mark  Looks good on identifyUser (https://share.commandbar.com/5KbhvP)
>   * white_check_mark Everything looks good for Segment's trackEvent (https://share.commandbar.com/m3D0fG)
> * I couldn't figure out how to use the old testing method to load CB, might be worth doing if easy `./bin/run serve --directory ./packages/browser-destinations/src/destinations --browser`.
> * Made a few minor copy nits

Thanks! 

> would it be possible/easy for us to add hmac in the scenario of `Deploy via Segment = false`?

Relatively easy I guess, but not completely trivial. We'd need a new SDK method (`addHmac` or similar) that works before and after boot. We'd then need to pick up the HMAC from wherever it was stored during boot. Maybe something we should add only if it's needed and a customer has a good reason not to use the currently supported approach?

**Comment by lchoward** [comment]
> Relatively easy I guess, but not completely trivial. We'd need a new SDK method (addHmac or similar) that works before and after boot. We'd then need to pick up the HMAC from wherever it was stored during boot. Maybe something we should add only if it's needed and a customer has a good reason not to use the currently supported approach?

Makes sense, thanks for the explanation. I agree with your take and leave as is for now (add later if needed).

Also checked on the old testing framework with Thomas's gentle nudge to check the testing instructions in the PR (😅 whoops) and it looks epic: https://share.commandbar.com/HmumYv 🚀 !

File: packages/browser-destinations/src/destinations/commandbar/types.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export type ModalFormFactor = { type: 'modal' }
+export type InlineFormFactor = { type: 'inline'; rootElement: string /* | HTMLElement */ }
```
Comment (vayyala) [inline packages/browser-destinations/src/destinations/commandbar/types.ts]: note we just fixed this [here](https://github.com/tryfoobar/monobar/pull/2250)

**Review by vayyala — APPROVED** [review]
Looks great! 👏

File: packages/browser-destinations/src/destinations/commandbar/types.ts (hunk 1 above)
Comment (tkainrad) [reply packages/browser-destinations/src/destinations/commandbar/types.ts]: Applied the new type also here now

**Pull request closed without merging**